packages/@okta/vuepress-theme-prose/components/ContentPage.vue

```diff
@@ -3,92 +3,101 @@
   data() {
     return {
       anchors: [],
-      headingAnchorsMap: {},
-      paddedHeaderHeight: 0
+      activeAnchor: null,
+      headingAnchorsMap: {}
     };
   },
   mounted() {
-    this.paddedHeaderHeight =
-      document.querySelector(".fixed-header").clientHeight +
-      LAYOUT_CONSTANTS.HEADER_TO_CONTENT_GAP;
+    this.anchors = this.getAnchors();
+
     if (document.readyState === "complete") {
-        this.scrollToActiveAnchor();
-        this.captureAnchors();
+      this.onURLAnchorChange();
     } else {
       window.addEventListener("load", () => {
-        this.scrollToActiveAnchor();
-        this.captureAnchors();
+        this.onURLAnchorChange();
+      });
+      window.addEventListener("popstate", e => {
+        e.target.location.hash && this.scrollToAnchor(e.target.location.hash);
       });
     }
+
+    window.addEventListener("scroll", this.onScrollHandleActiveHashListner);
+  },
+  beforeDestroy() {
+    window.removeEventListener("scroll", this.onScrollHandleActiveHashListner);
   },
   watch: {
     $page(to, from) {
-      this.$nextTick(function() {
-        this.scrollToActiveAnchor();
-        if (from.title !== to.title) {
-          this.captureAnchors();
-        }
-      });
+      if (from.title !== to.title) {
+        this.$nextTick(function() {
+          this.anchors = this.getAnchors();
+          this.getActiveAnchor(this.anchors, true);
+          this.onURLAnchorChange();
+        });
+      }
     }
   },
   methods: {
-    onAnchorClick(event) {
-      const element = event.target.hash
-        ? event.target
-        : event.target.closest("a");
-      if (
-        location.pathname.replace(/^\//, "") ==
-          element.pathname.replace(/^\//, "") &&
-        location.hostname == element.hostname
-      ) {
-        let scrollToAnchor = this.headingAnchorsMap[element.hash];
-        if (scrollToAnchor) {
-          event.preventDefault();
-          location.hash = element.hash;
-          return false;
-        }
+    onURLAnchorChange() {
+      const anchor = window.location.href.split("#")[1];
+
+      if (anchor) {
+        this.scrollToAnchor(`#${anchor}`);
+      } else {
+        // navigating via back button to no-anchor URL
+        window.scrollTo(0, 0);
       }
+      this.onClickCaptureAnchors();
     },
-    captureAnchors() {
+    onClickCaptureAnchors() {
       this.anchors.forEach(
         link => link.removeEventListener("click", this.onAnchorClick),
         this
       );
 
-      this.headingAnchorsMap = Array.from(
-        document.querySelectorAll(".header-anchor.header-link")
-      ).reduce(function(anchorsByHash, anchor) {
+      this.headingAnchorsMap = this.anchors.reduce((anchorsByHash, anchor) => {
         anchorsByHash[anchor.hash] = anchor;
         return anchorsByHash;
       }, {});
-      this.anchors = Array.from(
-        document.querySelectorAll(
-          'a[href^="#"]:not(.on-this-page-link):not(.tree-nav-link)'
```

Comment:
this might be still necessary to attach click handler to non-heading anchors
<img width="351" alt="Screenshot 2021-04-09 at 17 17 30" src="https://user-images.githubusercontent.com/71440851/114194180-e5b54a00-9957-11eb-9abe-15dd2950091b.png">